DQM/L1TMonitor/src/L1TObjectsTiming.cc

```diff
@@ -0,0 +1,480 @@
+    etsum_eta_phi_MET_firstbunch.reserve(bxrange);
+    etsum_eta_phi_METHF_firstbunch.reserve(bxrange);
+    etsum_eta_phi_MHT_firstbunch.reserve(bxrange);
+    etsum_eta_phi_MHTHF_firstbunch.reserve(bxrange);
+    muons_eta_phi_lastbunch.reserve(bxrange);
+    jet_eta_phi_lastbunch.reserve(bxrange);
+    egamma_eta_phi_lastbunch.reserve(bxrange);
+    tau_eta_phi_lastbunch.reserve(bxrange);
+    etsum_eta_phi_MET_lastbunch.reserve(bxrange);
+    etsum_eta_phi_METHF_lastbunch.reserve(bxrange);
+    etsum_eta_phi_MHT_lastbunch.reserve(bxrange);
+    etsum_eta_phi_MHTHF_lastbunch.reserve(bxrange);
+    
+}
+
+L1TObjectsTiming::~L1TObjectsTiming() {}
+
+void L1TObjectsTiming::fillDescriptions(edm::ConfigurationDescriptions& descriptions) {
+  edm::ParameterSetDescription desc;
+  desc.add<edm::InputTag>("muonProducer")->setComment("uGMT output muons.");;
+  desc.add<edm::InputTag>("stage2CaloLayer2JetProducer")->setComment("stage2 output jet");
+  desc.add<edm::InputTag>("stage2CaloLayer2EGammaProducer")->setComment("stage2 output egamma");
+  desc.add<edm::InputTag>("stage2CaloLayer2TauProducer")->setComment("stage2 output tau");
+  desc.add<edm::InputTag>("stage2CaloLayer2EtSumProducer")->setComment("stage2 output etsum");
+  desc.add<edm::InputTag>("ugtProducer")->setComment("uGT output");
+  desc.addUntracked<std::string>("monitorDir", "")->setComment("Target directory in the DQM file. Will be created if not existing.");
+  desc.addUntracked<bool>("verbose", false);
+  desc.addUntracked<std::string>("firstBXInTrainAlgo","")->setComment("Pick the right algo trigger bit for  L1 First Collision In Train");
+  desc.addUntracked<std::string>("lastBXInTrainAlgo","")->setComment("Pick the right algo trigger bit for  L1 Last Collision In Train");
+  descriptions.add("l1tObjectsTiming", desc);
+}
+
+void L1TObjectsTiming::dqmBeginRun(const edm::Run& r, const edm::EventSetup& c) {
+   // Get the trigger menu information
+    gtUtil_->retrieveL1Setup(c);
+    // Get the algo bits needed for the timing histograms
+    if (!gtUtil_->getAlgBitFromName(algoNameFirstBxInTrain_, algoBitFirstBxInTrain_)) {
+      edm::LogWarning("L1TObjectsTiming") << "Algo \"" << algoNameFirstBxInTrain_ << "\" not found in the trigger menu " << gtUtil_->gtTriggerMenuName() << ". Could not retrieve algo bit number.";
+    }
+    if (!gtUtil_->getAlgBitFromName(algoNameLastBxInTrain_, algoBitLastBxInTrain_)) {
+      edm::LogWarning("L1TObjectsTiming") << "Algo \"" << algoNameLastBxInTrain_ << "\" not found in the trigger menu " << gtUtil_->gtTriggerMenuName() << ". Could not retrieve algo bit number.";
+    }
+
+
+}
+
+void L1TObjectsTiming::beginLuminosityBlock(const edm::LuminosityBlock&, const edm::EventSetup&) {}
+
+void L1TObjectsTiming::bookHistograms(DQMStore::IBooker& ibooker, const edm::Run&, const edm::EventSetup&) {
+
+
+  // Subsystem Monitoring and Muon Output
+  std::string bx_obj[5]={"minus2","minus1","0","plus1","plus2"};
```

Comment:
Better use a `const std::array<std::string, 5> bx_obj = {"minus2","minus1","0","plus1","plus2"};`

In principle the size of the array is defined by the class member variable "bxrange". This probably throws an exception if you choose a bxrange > 5.
So even better would be to define this with a size of bxrange and set the strings from `"minus"+std::to_string(std::abs(std::min(0, 1 - (int)bxrange%2 - (int)std::floor(bxrange/2.))))` to `"plus"+std::to_string(std::max(0, (int)std::floor(bxrange/2.)))`.